shared/util.go

```diff
@@ -614,6 +614,31 @@ func StringInSlice(key string, list []string) bool {
 	return false
 }
 
+// RemoveElementsFromStringSlice returns a slice equivalent to removing the given elements from the given list.
+// If any elements are not present in the list, an error is returned. This includes repeated elements, e.g. removing
+// ["foo", "foo", "foo"] from ["foo", "foo"] will return an error.
+func RemoveElementsFromStringSlice(elements []string, list []string) ([]string, error) {
```

Comment:
Could you swap the argument order and then make `elements` a variadic so that this usage becomes possible:

```
RemoveElementsFromStringSlice([]string{"foo", "bar"}, "foo")
RemoveElementsFromStringSlice([]string{"foo", "bar"}, "foo", "bar")
RemoveElementsFromStringSlice([]string{"foo", "bar"}, []string{"foo", "bar"}...)
```

Reply:
Yep that would be nicer.

Reply:
Also, unless there's a specific reason I'm missing here, it would be nice to have it perform more like `trim` style functions and not return an error if the requested elements were not found. If we wanted to check for the presence of elements we could use `StringInSlice()`.

Reply:
The reason for this was to ensure duplicates are invalid, otherwise
```go
RemoveElementsFromStringSlice([]string{"foo"}, "foo", "foo")
```
would be valid even if we had validated each element to be removed with `StringInSlice` before the call.

Reply:
Is that a problem though, its effectively the same as calling:

```
RemoveElementsFromStringSlice([]string{"foo"}, "foo")
RemoveElementsFromStringSlice([]string{}, "foo")
```

Reply:
Hmm yeah I see your point, will change